src/spikeinterface/extractors/nwbextractors.py

```diff
@@ -166,209 +146,453 @@ def read_nwbfile(
+        different sources, such as local disk or remote servers.
+    load_channel_properties: bool, default: True
+        If True, all the channel properties are loaded from the NWB file and stored as properties.
+        For streaming purposes, it can be useful to set this to False to speed up streaming.
+    file: file-like object or None, default: None
+        A file-like object representing the NWB file. Use this parameter if you have an in-memory
+        representation of the NWB file instead of a file path.
+    cache: bool, default: False
+        Indicates whether to cache the file locally when using streaming. Caching can improve performance for
+        remote files.
+    stream_cache_path: str, Path, or None, default: None
+        Specifies the local path for caching the file. Relevant only if `cache` is True.
+    storage_options: dict | None = None,
+        Additional parameters for the storage backend (e.g. AWS credentials) used for "zarr" stream_mode.
+    use_pynwb: bool, default: False
+        Uses the pynwb library to read the NWB file. Setting this to False, the default, uses h5py
+        to read the file. Using h5py can improve performance by bypassing some of the PyNWB validations.
+
+    Returns
+    -------
+    recording : NwbRecordingExtractor
+        The recording extractor for the NWB file.
+
+    Examples
+    --------
+    Run on local file:
+
+    >>> from spikeinterface.extractors.nwbextractors import NwbRecordingExtractor
+    >>> rec = NwbRecordingExtractor(filepath)
+
+    Run on s3 URL from the DANDI Archive:
+
+    >>> from spikeinterface.extractors.nwbextractors import NwbRecordingExtractor
+    >>> from dandi.dandiapi import DandiAPIClient
+    >>>
+    >>> # get s3 path
+    >>> dandiset_id, filepath = "101116", "sub-001/sub-001_ecephys.nwb"
+    >>> with DandiAPIClient("https://api-staging.dandiarchive.org/api") as client:
+    >>>     asset = client.get_dandiset(dandiset_id, "draft").get_asset_by_path(filepath)
+    >>>     s3_url = asset.get_content_url(follow_redirects=1, strip_query=True)
+    >>>
+    >>> rec = NwbRecordingExtractor(s3_url, stream_mode="fsspec", stream_cache_path="cache")
     """
 
     extractor_name = "NwbRecording"
     mode = "file"
+    name = "nwb"
     installation_mesg = "To use the Nwb extractors, install pynwb: \n\n pip install pynwb\n\n"
 
     def __init__(
         self,
         file_path: str | Path | None = None,  # provide either this or file
-        electrical_series_name: str | None = None,
+        electrical_series_name: str | None = None,  # deprecated
         load_time_vector: bool = False,
         samples_for_rate_estimation: int = 1_000,
-        stream_mode: Optional[Literal["fsspec", "ros3", "remfile"]] = None,
+        stream_mode: Optional[Literal["fsspec", "ros3", "remfile", "zarr"]] = None,
         stream_cache_path: str | Path | None = None,
+        electrical_series_path: str | None = None,
```

Comment:
to kwargs also.

Reply:
I would move this at the same place as `electrical_series_name`

Reply:
You mean in the signature? I argued the opposite because if someone was relying on the order of position arguments it will break their code.